### ISSUE TITLE
Display Lag when Screen Sharing

### ISSUE THREAD

**Issue by astralchan** [swaywm/sway#7162]
- **Sway Version:**
sway version 1.7

- **Debug Log:** (`sway -dV > sway.log 2>&1`)
[sway.log](https://gist.github.com/astralchan/ab99698eb6cd550afc6867f0b949d390)

- **Configuration File:**
[config](https://gist.github.com/astralchan/59d881c17ff8771faa997a95b1138f72)

- **Stack Trace:**
N/A (not a crash)

- **Additional Info:**
  - OS: Gentoo (x86_64) musl-hardened overlay
  - GPU: Nvidia Geforce GTX 1080
  - Driver: nouveau (`x11-drivers/xf86-video-nouveau-1.0.17`)
  - `gui-wm/sway` `USE` flags: `tray wallpapers`
  - pipewire version: `media-video/pipewire-0.3.56`
  - xdg-desktop-portal version: `sys-apps/xdg-desktop-portal-0.14.6`
  - xdg-desktop-portal-wlr version: `gui-libs/xdg-desktop-portal-wlr-0.5.0`
  - How sway is started:
```sh
#!/bin/sh

export WLR_NO_HARDWARE_CURSORS=1
export QT_QPA_PLATFORM=wayland
export SDL_VIDEODRIVER=wayland
export XDG_SESSION_TYPE=wayland
dbus-run-session sway -dV
```

- **Description:**
Whenever I screenshare, the window is extremely slow. Input is lagged and key presses are duplicated. Whenever the screenshare is stopped, performance is back to normal.

- **Attempted Fixes:**
  - Issue [7118](https://github.com/swaywm/sway/issues/7118) looks to be due to unsupported driver. However, my card (nv130) is listed in the [nouveau feature matrix](https://nouveau.freedesktop.org/FeatureMatrix.html) as supported. Plus, I don't see any errors with that in the log above.
  - Issue [5610](https://github.com/swaywm/sway/issues/5610) describes an issue with a card without support. The nv130 card I use is supported.
  - Issue [82](https://github.com/swaywm/sway/issues/82) setting `WLC_BG` to `0` does nothing

**Comment by zsolt-donca** [comment]
I know that this is not a fix, but you could try setting a low value to the `max_fps` setting of [xdg-desktop-portal-wlr](https://man.archlinux.org/man/xdg-desktop-portal-wlr.5). It could help with input lag (see https://github.com/emersion/xdg-desktop-portal-wlr/issues/66).

**Comment by astralchan** [comment]
> I know that this is not a fix, but you could try setting a low value to the `max_fps` setting of [xdg-desktop-portal-wlr](https://man.archlinux.org/man/xdg-desktop-portal-wlr.5). It could help with input lag (see [emersion/xdg-desktop-portal-wlr#66](https://github.com/emersion/xdg-desktop-portal-wlr/issues/66)).

putting

```
[screencast]
outputname=
max_fps=10
```

into `~/.config/xdg-desktop-portal-wlr/config` doesn't change anything. It might also be worth noting that the same lag occurs with the grim screenshot utility.

**Comment by rjonczy** [comment]
@astralchan I am wondering if you were able to fix that somehow? I am having similar issue.

**Comment by astralchan** [comment]
> @astralchan I am wondering if you were able to fix that somehow? I am having similar issue.

No, I was never ever to solve the issue. I ended up running a qemu / virt-manager instance of windows 11 and capturing from there.

**Comment by JoshuaKir** [comment]
@rjonczy Any chance you found a fix? I've been stumped on this for hours

**Comment by JoshuaKir** [comment]
Hi @zsolt-donca, I am experiencing the same issue except I am running more recent portals and pipewire, and I am on AMD hardware. Would you recommend I open a separate issue on the wlr portal page or is there an ongoing issue I can tag along with?

**Comment by edu-vi** [comment]
Same issue with hyperland btw, amd hardware

**Comment by JoshuaKir** [comment]
> Same issue with hyperland btw, amd hardware

Forgot to give an update! I didnt have a cpu freq governor set up and i also installed GameMode. This seems to have fixed my issues. unfortunately im in the process of moving right now so i dont have my computer set up to give better details. Hope this helps, best of luck!
